k8sutils/pkg/workload/ownerreference.go

```diff
@@ -56,43 +56,30 @@ func GetWorkloadFromOwnerReference(ownerReference metav1.OwnerReference) (worklo
 }
 
 func GetWorkloadNameAndKind(ownerName, ownerKind string) (string, k8sconsts.WorkloadKind, error) {
-	if ownerKind == "ReplicaSet" {
-		return extractDeploymentInfo(ownerName)
+	switch ownerKind {
+	case "ReplicaSet":
+		return extractInfoWithSuffix(ownerName, k8sconsts.WorkloadKindDeployment)
+	case "Job":
+		return extractInfoWithSuffix(ownerName, k8sconsts.WorkloadKindCronJob)
+	default:
+		return extractInfoWithoutSuffix(ownerName, ownerKind)
```

Comment:
could be nice to document here your findings on the format of each case, what the "suffix" means in this case (not straight forward what it means) :)

also could be nice to add a comment that the name can contain multiple hyphens so in the future we would know this is an issue and to notice if we make any refactors here